src/lib/tests/tests.cpp

```diff
@@ -1,6 +1,399 @@
+#include "simd/simd_doubles.h"
+#include "simd/simd_floats.h"
+#include "simd_config.h"
 #include "gtest/gtest.h"
 
-// a blank test
-TEST(DistopiaTests, BlankTest) {
-  SUCCEED();
+/* Section for testing SimdFloatX4 packed structs
+Notes
+* operates on __m128 types which require 16 bit alignment
+
+*/
+
+TEST(SimdFloatX4, SetAndStore) {
+  SimdFloatX4 fx4;
+  float *result = new float[4];
+  fx4.set(1.0F);
+  fx4.storeU(result);
+  for (size_t i = 0; i < 4; i++) {
+    EXPECT_FLOAT_EQ(result[i], 1.0F);
+  }
+  delete[] result;
+}
+
+/* __m128 requires 16 bit alignment */
+TEST(SimdFloatX4, AlignedLoadAndStore) {
+  SimdFloatX4 fx4;
+  float vals[4] __attribute__((aligned(16))) = {1.0F, 2.0F, 3.0F, 4.0F};
+  float *result = static_cast<float *>(aligned_alloc(16, 4 * sizeof(float)));
+  fx4.load(vals);
+  fx4.store(result);
+  for (size_t i = 0; i < 4; i++) {
+    EXPECT_FLOAT_EQ(result[i], vals[i]);
+  }
+  delete[] result;
+}
+
+TEST(SimdFloatX4, UnalignedLoadAndStore) {
+  SimdFloatX4 fx4;
+  float vals[4]{1.0F, 2.0F, 3.0F, 4.0F};
+  float *result = new float[4];
+  fx4.loadU(vals);
+  fx4.storeU(result);
+  for (size_t i = 0; i < 4; i++) {
+    EXPECT_FLOAT_EQ(result[i], vals[i]);
+  }
+  delete[] result;
+}
+
+TEST(SimdFloatX4, Zero) {
+  SimdFloatX4 fx4;
+  float zeros[4] = {0.0F, 0.0F, 0.0F, 0.0F};
+  float *result = new float[4];
+  fx4.zero();
```

Comment:
This test will pass even if `.zero` and `.storeU` do nothing whatsoever because you initialise `zeros` to the correct value~

Reply:
oops....